osgar/logger.py

```diff
@@ -58,18 +58,38 @@ def format_header(start_time):
 
 def format_packet(stream_id, data, dt):
     ret = []
-    assert dt.days == 0, dt  # multiple days not supported yet
-    time_frac = (dt.seconds * 1000000 + dt.microseconds) & TIMESTAMP_MASK
+    dt_bytes = format_timedelta(dt)
     index = 0
     while index + 0xFFFF <= len(data):
-        ret.append(struct.pack('IHH', time_frac, stream_id, 0xFFFF))  # header
+        ret.append(dt_bytes + struct.pack('HH', stream_id, 0xFFFF))  # header
         ret.append(data[index:index + 0xFFFF]) # partial data
         index += 0xFFFF
-    ret.append(struct.pack('IHH', time_frac, stream_id, len(data) - index))  # header
+    ret.append(dt_bytes + struct.pack('HH', stream_id, len(data) - index))  # header
     ret.append(data[index:])
     return ret
 
 
+def format_timedelta(dt):
+    assert dt.days == 0, dt  # multiple days not supported yet
+    time_frac = (dt.seconds * 1000000 + dt.microseconds) & TIMESTAMP_MASK
+    return time_frac.to_bytes(4, 'little')
```

Comment:
I am not sure if it is good idea to switch alternative coding, if otherwise `struct.pack` is used.

Reply:
Is that the only comment or are you still reviewing the rest?

Reply:
No, that was only a "quick comment"/first impression. I wanted to check the compatibility and the motivation for this PR, but I was too tired for this (2days ago).

Reply:
How can I help?